docs/architecture/adr-110-remote-mempool.md

```diff
@@ -0,0 +1,364 @@
+  application-related concerns (as per [RFC 011]). Should remote mempool
+  developers want to limit returned transactions by gas, this should be
+  implemented as part of the configuration of the remote mempool.
+
+### Impact and properties
+
+A new mempool that implements the [`Mempool`] interface will be created, and
+when enabled it will have the following impacts/properties:
+
+1. The mempool-related ABCI methods will not be called.
+2. The following methods will do nothing:
+   - `CheckTx`, since transaction validation is assumed to happen prior to
+     transaction insertion in the mempool.
+   - `FlushAppConn`, which is only relevant for use via ABCI.
+   - `RemoveTxByKey`, which is only used in the callback in the context of ABCI.
+   - `ReapMaxTxs`, which is only used internally and by the `unconfirmed_txs`
+     RPC endpoint.
+   - `Flush`, which is only ever used via the `unsafe_flush_mempool` RPC
+     endpoint.
+   - `Size` and `SizeBytes`, since these are only used for informational
+     purposes in the RPC.
+3. Reaping of transactions will result in network calls to the remote mempool
+   instance.
+4. The following RPC endpoints will be disabled, returning relevant errors when
+   users attempt to call them:
+   - `broadcast_tx_*`
+   - `unconfirmed_txs`
+   - `num_unconfirmed_txs`
+   - `unsafe_flush_mempool`
+5. All P2P-based transaction broadcast functionality will be disabled. It will
+   be up to the mempool developer to implement connectivity to other mempool
+   instances and the relevant broadcast/gossip mechanisms to disseminate
+   incoming transactions.
+
+### Startup and failure modes
+
+- When a CometBFT-based validator starts up with `mempool.type` set to `remote`,
+  it will expect to be able to connect to the remote mempool. If it cannot, it
+  will panic.
+
+- When a validator attempts to call any of the `Fetch` or `Reap` methods and the
+  call fails for whatever reason, it will panic.
+
+- It must be kept in mind that transactions will persist in the mempool between
+  CometBFT node restarts. This implies a different set of assumptions as
+  compared to when using the default mempool, as when a node fails with the
+  default mempool, the contents of the mempool are automatically flushed.
+
+### Rollout plan
+
+The remote mempool will initially land in CometBFT v1.0, and will potentially be
+considered for backporting to the `v0.38.x-experimental` branch.
+
+## Consequences
+
+### Positive
+
+- Application developers can provide their own mempools that live in a separate
+  process to the validator in which they can implement properties that benefit
+  their application specifically.
+- Transactions can be submitted to processes outside of a validator without
+  needing to run a full node or sentry, potentially reducing operational costs.
+- Separating the mempool out into its own process, if the mempool is built
+  correctly, can allow it to scale independently of the consensus engine.
+
```

Comment:
also it should be possible to implement Solana style (Gulf Stream) transaction forwarding